src/bigquery.ts

```diff
@@ -807,12 +873,18 @@ export class BigQuery extends common.Service {
    * @returns {object} A properly-formed `queryParameter` object.
    */
   // tslint:disable-next-line no-any
-  static valueToQueryParameter_(value: any) {
+  static valueToQueryParameter_(value: any, providedType: any = undefined) {
```

Comment:
Is `providedType` really `any`?  I'm surprised we can't further constrain this to `{}`.  Side note - for `undefined`, you don't need to set a default value.  You can say `providedType?: {}`

Reply:
providedType can be a string, object, or array, so I don't understand how to make providedType?: {} work this way.

Reply:
you could do something along the lines of:

```
providedType: string|{[key: string]: string}|string[]
```

:point_up: `|` is used to allow for multiple types.

Reply:
@bcoe The array type can also be populated with string, array, or object. How do I specify that without using any?

Reply:
@steffnay you'd potentially probably want to start breaking it out into interfaces, but you'd do something along the lines of:

```
{[key: string]: string|string[]}
```

:point_up: but as it gets complex like this, you'd probably want to start describing an interface. The advantage of putting in this work is that folks using the API have more safety, and it's easier to figure out what to pass to the API.